src/VisualStudio/Core/Def/Implementation/Remote/RemoteHostClientServiceFactory.RemoteHostClientService.cs

```diff
@@ -57,18 +54,14 @@ public void Enable()
                         return;
                     }
 
-                    if (!_workspace.Options.GetOption(RemoteHostOptions.RemoteHost))
+                    // We enable the remote host if either RemoteHostTest or RemoteHost are on.
+                    if (!_workspace.Options.GetOption(RemoteHostOptions.RemoteHostTest) &&
```

Comment:
I've opened https://github.com/dotnet/roslyn/pull/14215 to track removing the extra option.